web/src/components/Configuration/Configuration1.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import './Configuration.css'
+class Configuration1 extends Component{
+    constructor(props){
+        super(props);
+        this.state = {  
+           "openmp.threads": 4,
+            "mrtrix.threads": 2,
+            aseg_lh_labels : "8 10 11 12 13 16 17 18 26",
+            aseg_rh_labels:"47 49 50 51 52 53 54 58",
+            use_flirt:"True",
+            strmlns_no:"2M",
+            strmlns_sift_no:"1M",
+            strmlns_len:250,
+            strmlns_step:0.5,
+            "ct.elec.intensity.th":1000,
+            "decim.factor":0.1
+        }
+    }
+    handleChangeAll = (event) =>{
+        this.setState ( { [event.target.name] :event.target.value  } )
+       }
+       
+    onClickHandler=(event)=>{
+        alert("Data Saved for Patient 1")
+ event.preventDefault();
+ const label1=this.state["openmp.threads"]
+ const label2=this.state["mrtrix.threads"]
+ const label3=this.state["ct.elec.intensity.th"]
+ const label4=this.state["decim.factor"]
+ const aseg_lh_labels=this.state.aseg_lh_labels
+ const aseg_rh_labels=this.state.aseg_rh_labels
+ const use_flirt=this.state.use_flirt
+ const strmlns_no=this.state.strmlns_no
+ const strmlns_sift_no=this.state.strmlns_sift_no
+ const strmlns_len=this.state.strmlns_len
+ const strmlns_step=this.state.strmlns_step
+ const data={label1,label2,label3,label4,aseg_lh_labels,aseg_rh_labels,use_flirt,strmlns_no,strmlns_sift_no,strmlns_len,strmlns_step}
+ axios.post("http://localhost:8000/input1", data)
+.then(res => {
+  console.log(res.statusText)
+})
+    }
+    render(){
+        return(
+            <div className="container">
+            
+                <form className="config_form">
+                 <h1>Configuration for Patient 1 </h1>
+       
+     <label className="l_name"> openmp.threads 
+    <input  type="number" name="openmp.threads"  className="field" value={this.state["openmp.threads"]} onChange={this.handleChangeAll} />
+    <i  class="fa fa-info-circle fa-1x" title="Default values is 4.Determine no of threads in multi-processing." ></i>
+ 
+    </label>
```

Comment:
This set of tags (label + input + i) can be moved to a generic component called InputField that receives as arguments the label ('openmp.threads'), the value ({this.state["openmp.threads"]}) and the title description.
Once we have such a method, we could transform all the code from lines 52-106 into a loop that iterates over the state and generates an InputField component with the proper label and title.